pertpy/tools/_mixscape.py

```diff
@@ -0,0 +1,225 @@
+from __future__ import annotations
+
+import numpy as np
+import pandas as pd
+import scanpy as sc
+from anndata import AnnData
+from scipy import sparse
+from sklearn.mixture import GaussianMixture
+
+
+def _define_normal_mixscape(X: np.ndarray | sparse.spmatrix | pd.DataFrame | None) -> list[float]:  # noqa: N803
+    """Calculates the mean and standard deviation of a matrix.
+
+    Args:
+        X: The matrix to calculate the properties for.
+
+    Returns:
+        Mean and standard deviation of the matrix.
+    """
+    mu = X.mean()
+    sd = X.std()
+
+    return [mu, sd]
+
+
+def mixscape(
+    adata: AnnData,
+    pert_key: str,
+    labels: str,
+    control: str,
+    new_class_name: str | None = "mixscape_class",
+    min_de_genes: int | None = 5,
+    min_cells: int | None = 5,
+    layer: str | None = None,
+    logfc_threshold: float | None = 0.25,
+    iter_num: int | None = 10,
+    split_by: str | None = None,
+    pval_cutoff: float | None = 5e-2,
+    fine_mode: bool | None = False,
+    fine_mode_labels: str | None = "guide_ID",
+    prtb_type: str | None = "KO",
+    copy: bool | None = False,
```

Comment:
Maybe a better idea to only allow bool -> True/False and set the default to False? I don't see why a None value would make sense here.